controllers/cinder_controller.go

```diff
@@ -630,31 +629,50 @@ func (r *CinderReconciler) reconcileNormal(ctx context.Context, instance *cinder
 		instance.Status.Conditions.Set(c)
 	}
 
-	// deploy cinder-backup
-	cinderBackup, op, err := r.backupDeploymentCreateOrUpdate(ctx, instance)
-	if err != nil {
-		instance.Status.Conditions.Set(condition.FalseCondition(
-			cinderv1beta1.CinderBackupReadyCondition,
-			condition.ErrorReason,
-			condition.SeverityWarning,
-			cinderv1beta1.CinderBackupReadyErrorMessage,
-			err.Error()))
-		return ctrl.Result{}, err
-	}
-	if op != controllerutil.OperationResultNone {
-		r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
-	}
+	// deploy cinder-backup, but only if necessary
+	//
+	// Many OpenStack deployments don't use the cinder-backup service (it's optional),
+	// so there's no need to deploy it unless it's required.
+	var backupCondition *condition.Condition
+	if instance.Spec.CinderBackup.Replicas > 0 {
+		cinderBackup, op, err := r.backupDeploymentCreateOrUpdate(ctx, instance)
+		if err != nil {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				cinderv1beta1.CinderBackupReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityWarning,
+				cinderv1beta1.CinderBackupReadyErrorMessage,
+				err.Error()))
+			return ctrl.Result{}, err
+		}
+		if op != controllerutil.OperationResultNone {
+			r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
+		}
 
-	// Mirror CinderBackup status' ReadyCount to this parent CR
-	instance.Status.CinderBackupReadyCount = cinderBackup.Status.ReadyCount
+		// Mirror CinderBackup status' ReadyCount to this parent CR
+		instance.Status.CinderBackupReadyCount = cinderBackup.Status.ReadyCount
 
-	// Mirror CinderBackup's condition status
-	c = cinderBackup.Status.Conditions.Mirror(cinderv1beta1.CinderBackupReadyCondition)
-	if c != nil {
-		instance.Status.Conditions.Set(c)
+		// Mirror CinderBackup's condition status
+		backupCondition = cinderBackup.Status.Conditions.Mirror(cinderv1beta1.CinderBackupReadyCondition)
+
+	} else {
+		// Clean up cinder-backup if there are no replicas
+		err = r.backupCleanupDeployment(ctx, instance)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	if backupCondition != nil {
+		// If there was a Status=False condition, set that as the CinderBackupReadyCondition
```

Comment:
It's not necessarily the case that the condition here is `Status=False`.  Maybe we should change the comment to remove that part?